src/surface/photosphere/fisheye_projection.py

```diff
@@ -0,0 +1,148 @@
+import math
+from dataclasses import dataclass
+
+import cv2
+import numpy as np
+
+
+@dataclass
+class FisheyeImage:
+    img: np.ndarray
+    img_num: int
+    left: int
+    top: int
+    diameter: int
+
+
+# The dimensions of the output image in width by height
+OUTPUT_DIMENSION = (2000, 1000)
+
+# The APERTURE of the fisheyes in radians
+APERTURE = 195 * math.pi / 180
+
+# The fisheye images with their information
+FISHEYE_IMAGES = [
+    FisheyeImage(
+        img=cv2.imread('src/surface/photosphere/fisheye1.jpg'),
+        img_num=0,
+        left=400,
+        top=28,
+        diameter=3052,
+    ),
+    FisheyeImage(
+        img=cv2.imread('src/surface/photosphere/fisheye2.jpg'),
+        img_num=1,
+        left=404,
+        top=-25,
+        diameter=3040,
+    ),
+]
+
+
+# Converts the coordinates in the projection to coordinates in the fisheye image
+# all coordinates are in unit coordinates (-1 to 1 in all dimensions with 0 being center)
+def projection_to_fisheye(projection_coord: list, img: int) -> tuple:
+    # Equirectangular to latitude/longitude
+    theta = math.pi * projection_coord[0] + math.pi * img
+    phi = math.pi * projection_coord[1] / 2
+
+    # Latitude/Longitude to 3D vector
+    px = math.cos(phi) * math.sin(theta)
+    py = math.cos(phi) * math.cos(theta)
+    pz = math.sin(phi)
+
+    # 3D vector to 2D fisheye
+    r = 2 * math.atan2(math.sqrt(px * px + pz * pz), py) / APERTURE
+    theta = math.atan2(pz, px)
+
+    fisheye_col = r * math.cos(theta)
+    fisheye_row = r * math.sin(theta)
+
+    return (fisheye_row, fisheye_col)
+
+
+# Calculate the unit coordinate according to the given normal coordinate and width
+def normal_to_unit_grid(x: int, width: int) -> float:
+    return (x / width - 0.5) * 2
+
+
+# Calculate the normal coordinate according to the given unit coordinate and width
+def unit_to_normal_grid(x: float, width: int) -> int:
+    return math.floor((x + 1) * width / 2)
+
+
+def unit_to_fisheye_coord(unit_coord: tuple, fisheye_num: int) -> tuple:
+    return (
+        unit_to_normal_grid(unit_coord[0], FISHEYE_IMAGES[fisheye_num].diameter)
+        + FISHEYE_IMAGES[fisheye_num].top,
+        unit_to_normal_grid(unit_coord[1], FISHEYE_IMAGES[fisheye_num].diameter)
+        + FISHEYE_IMAGES[fisheye_num].left,
+    )
+
+
+# The maximum width a single projection covers in unit coordinates
+MAX_WIDTH = APERTURE / 2 / math.pi
+
+# The portions of the projections that overlap and can be blurred
+LEFT_SEAM = [
+    unit_to_normal_grid(-1 * MAX_WIDTH, OUTPUT_DIMENSION[0]),
+    unit_to_normal_grid(-1 + MAX_WIDTH, OUTPUT_DIMENSION[0]),
+]
+RIGHT_SEAM = [
+    unit_to_normal_grid(1 - MAX_WIDTH, OUTPUT_DIMENSION[0]),
+    unit_to_normal_grid(MAX_WIDTH, OUTPUT_DIMENSION[0]),
+]
+
+if __name__ == '__main__':
```

Comment:
This should probably actually be a method so we can call it later once it's integrated. Something like:
```python
def equirectangular_projection(images):
    ...

if __name__ == '__main__':
    # make fisheye_images here or in equirectangular_projection?
    images = # Make fisheye_images
    projection = equirectangular_projection(images)
    cv2.imwrite('...', projection)
```